### PR TITLE
Add test cases to verify mt-MT(Maltese-Malta)

### PR DESCRIPTION
**Pull request by gouniLee**
### Checklist

* [ ] At least one test case is included for this feature or bug fix.
* [x] `ReleaseNotes` has added or is not needed.
* [x] This PR has passed all of the test cases on `Nodejs` and `Chrome Browser`.
* [x] This PR has passed all of the test cases on `QT/QML`.
* [ ] This is an API breaking change.
* [ ] Requires a major version change.

### Issue Resolved / Feature Added
* Added test cases to verify mt-MT (Maltese-Malta)
  * timezone: Europe/Malta
  * Added test case feature: LocaleInfo,Script, String(plural) LocaleMatch, DateFmt, DurFmt, DateRngFmt, List, Unit, firstDayofWeek, clock, calendar,Month/Weekday Translation, NumFmt,

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)

### Links
[//]: # (Related issues, references)

### CODE REVIEW

File: js/test/durfmt/testdurfmt2.js [hunk 1]
```diff
@@ -9443,6 +9443,71 @@ module.exports.testdurfmt2 = {
...
+            textformatted_14.push(textfmt.format({year: 14,month: 14,week: 14,day: 14}).toString());
+            textformatted_100.push(textfmt.format({year: 100, month: 100, week: 100, day: 100}).toString());
+
+            clockformatted_1.push(textfmt.format({hour: 1, minute: 1, second: 1}).toString());
+            clockformatted_5.push(textfmt.format({hour: 5, minute: 5, second: 5}).toString());
+            clockformatted_14.push(textfmt.format({hour: 14, minute: 14, second: 14}).toString());
+            clockformatted_100.push(textfmt.format({hour: 100, minute: 100, second: 100}).toString());
+        }
+
+        test.equal(textformatted_1[0], '1 y, 1 m, 1 w, u 1 d');
```
Comment (ehoogerbeets) [inline js/test/durfmt/testdurfmt2.js:9471]: Translations are available for Maltese on Google translate as well:

1 second = 1 sekonda
2 seconds = 2 sekondi
1 minute = 1 minuta
2 minutes = 2 minuti
1 hour = 1 siegħa
2 hours = 2 sigħat
1 day = 1 jum
2 days = Jumejn
3 days = 3 ijiem
1 week = Ġimgħa
2 weeks = Ġimagħtejn
3 weeks = 3 ġimgħat
1 month = Xahar
2 months = Xahrejn
3 months = 3 xhur
1 year = Sena
2 years = Sentejn
3 years = 3 snin

File: js/test/root/teststrings.js [hunk 2]
```diff
@@ -3590,6 +3590,43 @@ module.exports.teststrings = {
         test.equal(str.formatChoice(15), "Default items");
         test.done();
     },
+    testStringFormatChoiceCharClassesComplex_mt_MT: function(test) {
+        test.expect(2);
+        var str = new IString("0#There are no items.|one#There items are one|few#The items are few|many#The items are many|#Default items");
+        str.setLocale("mt-MT");
```
Comment (ehoogerbeets) [inline js/test/root/teststrings.js:3596]: According to http://web.stanford.edu/group/cslipublications/cslipublications/TLS/TLS9-2005/TLS9_David_Karine.pdf Maltese has singular (one), dual (two), and plural (other) categories.

File: js/test/strings-ext/testlistfmt_mt_MT.js [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+    },
+    testListFmtORStyleNumberFormatThree_mt_MT: function(test) {
+        var fmt = new ListFmt({
+            locale: "mt-MT",
+            style: "disjunction"
+        });
+
+        test.expect(2);
+        test.ok(fmt !== null);
+        test.equal(fmt.format(["waħda", "tnejn", "tlieta"]), "waħda, tnejn, or tlieta");
```
Comment (ehoogerbeets) [inline js/test/strings-ext/testlistfmt_mt_MT.js:228]: according to google translate, the translation of "or" in Maltese is "jew"

File: js/test/units/testunitfmt_mt_MT.js [hunk 4]
```diff
@@ -0,0 +1,331 @@
...
+        });
+
+        var uf = new UnitFmt({
+            locale: "mt-MT", 
+            autoConvert: true,
+            length: "long"
+        });
+
+        var str = uf.format(m1);
+        test.equal(str, "2 cm²");
```
Comment (ehoogerbeets) [inline js/test/units/testunitfmt_mt_MT.js:122]: Translations of the units are available on google translate

File: js/test/root/teststrings.js (hunk 2 above)
Comment (gouniLee) [reply js/test/root/teststrings.js:3596]: the locale data have generated based on https://unicode-org.github.io/cldr-staging/charts/37/supplemental/language_plural_rules.html there' is `many` category

File: js/test/strings-ext/testlistfmt_mt_MT.js (hunk 3 above)
Comment (gouniLee) [reply js/test/strings-ext/testlistfmt_mt_MT.js]: I've manually updated a list.json file to support it. CLDR says `or`

File: js/data/locale/mt/unitfmt.json [hunk 5]
```diff
@@ -8,10 +8,10 @@
             "degree": "#{n}°",
             "arc-minute": "#{n}′",
             "arc-second": "#{n}″",
-            "square-kilometer": "#{n} km²",
+            "square-kilometer": "#{n} kilometru kwadru",
```
Comment (ehoogerbeets) [inline js/data/locale/mt/unitfmt.json:11]: These need to be Maltese plurals. They should have different cases for the different plural categories.

File: js/data/locale/mt/list.json [hunk 6]
```diff
@@ -17,6 +17,20 @@
             "end": "{0}, u {1}"
         }
     },
+    "or": {
+        "medium": {
+            "2": "{0} jew {1}",
+            "end": "{0}, jew {1}"
```
Comment (ehoogerbeets) [inline js/data/locale/mt/list.json:23]: This special case needs to go into the cldr tool that generates the list formats (genlist.js) as a hard-coded format. That way, when we rerun the list format tool, we don't overwrite this special case.

File: js/data/locale/mt/unitfmt.json (hunk 5 above)
Comment (gouniLee) [reply js/data/locale/mt/unitfmt.json:11]: Yes, the problem is, Maltese have several plural categories. one, few, many, other.
so the format data form would have like this,
`one#{n} |few#{n} |many#{n} |#{n} `

 but I'm don't get how to get 'few` category translation correctly. so I'll just update the same with many
`one#{n}kilometru kwadru |few#{n} kilometri kwadri|many#{n}kilometri kwadri |#{n}kilometri kwadri `

square kilometer: kilometru kwadru
square killometers :  kilometri kwadri

Comment (ehoogerbeets) [reply js/data/locale/mt/unitfmt.json:11]: Enter the following into google:

1 square kilometer = Kilometru kwadru
2 square kilometers = 2 kilometri kwadri
14 square kilometers = 14-il kilometru kwadru
21 square kilometers = 21 kilometru kwadru

This translates to the following plural string:

"one#Kilometru kwadru|few#{n} kilometri kwadri|many#{n}-il kilometru kwadru|other#{n} kilometru kwadru"

File: js/test/durfmt/testdurfmt2.js (hunk 1 above)
Comment (gouniLee) [reply js/test/durfmt/testdurfmt2.js:9471]: As you wrote above, When I check the google translate, it looks like they have `two` category ;( 
2 days = Jumejn
3-10 days = {n} ijiem
14 days = {n}-il jum
100 days = {n} jum

Probably I need to report to CLDR.

**Review by ehoogerbeets — APPROVED** [review]

**Comment by gouniLee** [comment]
CLDR ticket for plural category : https://unicode-org.atlassian.net/browse/CLDR-14665

**Pull request merged**